### ISSUE TITLE
EdDSA PEM key is malformed

### ISSUE THREAD

**Issue by yaronf** [openssl/openssl#17644]
TL;DR: **Ed25519 PEM key is not compliant with the PKCS#8 definition.**

OpenSSL v3.0.1

```
$ openssl genpkey -algorithm Ed25519 -out ed25519key.pem
$ openssl asn1parse -in ed25519key.pem
```
I would expect the output to be compliant with [PKCS#8](https://datatracker.ietf.org/doc/html/rfc5208#appendix-A), with the private key a straight 32-byte value, as per RFC 8032. However:
```
openssl asn1parse -in prv.pem       
    0:d=0  hl=2 l=  46 cons: SEQUENCE          
    2:d=1  hl=2 l=   1 prim: INTEGER           :00
    5:d=1  hl=2 l=   5 cons: SEQUENCE          
    7:d=2  hl=2 l=   3 prim: OBJECT            :ED25519
   12:d=1  hl=2 l=  34 prim: OCTET STRING      [HEX DUMP]:04209F8362F87A484A954E6E740C5B4C0E84229139A20AA8AB56FF66586F6A7D29C5
 ```

The last element is a 34-byte long octet string. Surprisingly, the first 2 bytes of the string are 0x04 (the DER tag for “octet string”) and 0x20 (the length of the remaining bytes, and the correct key length). So OpenSSL is nesting one octet string within another. This is strange and IMHO differs from PKCS#8.
 
See also this comment: https://stackoverflow.com/a/60317259

Thanks!
